src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -73,18 +98,50 @@ public CommandResult execute(Model model) throws CommandException {
         if (index.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
-
         Person personToEdit = lastShownList.get(index.getZeroBased());
-        Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
 
-        if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
-            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
-        }
+        if (personToEdit instanceof Student) {
+
+            Student editedStudent;
+            Student studentToEdit = (Student) personToEdit;
+            if (editStudentDescriptor != null) {
+                editedStudent = createEditedStudent(studentToEdit, editStudentDescriptor);
+            } else {
+                EditStudentDescriptor editStudentDescriptor = new EditStudentDescriptor(editPersonDescriptor);
+                editedStudent = createEditedStudent(studentToEdit, editStudentDescriptor);
+            }
+
+            // if you are changing the name to one that already exists
+            if (!studentToEdit.isSamePerson(editedStudent) && model.hasPerson(editedStudent)) {
+                throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+            }
+
+            if (!model.hasTutorialWithName(editedStudent.getTutorialName())) {
+                throw new CommandException(String.format(MESSAGE_TUTORIAL_DOES_NOT_EXIST,
+                        editedStudent.getTutorialName()));
+            }
+
+            model.setPerson(studentToEdit, editedStudent);
+            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
+            return new CommandResult(String.format(MESSAGE_EDIT_STUDENT_SUCCESS, editedStudent));
 
-        model.setPerson(personToEdit, editedPerson);
-        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        } else {
+            if (editStudentDescriptor != null) {
+                throw new CommandException(MESSAGE_NOT_A_STUDENT);
```

Comment:
the GradeCommand also has a similar message, would be good if we can abstract it out into the common.core.Messages class!

Reply:
I think the current message is more suitable because it is shown when the user tries to change the `NusNetId` or the `TutorialName` , however the message in gradeCommand says there is no student with the given `Name`. so i think the current one is more suitable. Although u are right, ill try to abstract out some of the messages